[breaking] Fix board attach CLI command, and remove gRPC API (#1930)
arduino/arduino-cli: docs/UPGRADING.md

Here you can find a list of migration guides to handle breaking changes between releases of the CLI.

## Unreleased

### `board attach` CLI command and `BoardAttach` gRPC interface command removal

Both the `board attach` CLI command and the `cc.arduino.cli.commands.v1.BoardAttach` gRPC command have been removed.

> They are not replaced by any new command, the only way to get the same behaviour is to edit the `sketch.json` file either manually or automatically with other tools.

Reviewer: Is it true? is `sketch.json` supported in any way after removing `board attach`?

The errors are automatically sent to output via `feedback` package, as for the other `Init*` functions.

### `board attach` CLI command changed behaviour

> The `board attach` CLI command has changed behaviour: now it just pick whatever port and FQBN is passed as parameter and
> saves it in the `sketch.json` file, without any validity check or board autodetection.

Reviewer 2: ```suggestion
The `board attach` CLI command has changed behaviour: now it just picks whatever port and FQBN is passed as parameter
and saves it in the `sketch.json` file, without any validity check or board autodetection.
```

Minor grammar fix.

> saves it in the `sketch.json` file, without any validity check or board autodetection.

Reviewer 3: needs fixing

Reviewer: I changed it in the next commits.

### `cc.arduino.cli.commands.v1.BoardAttach` gRPC interface command removal

> The `cc.arduino.cli.commands.v1.BoardAttach` gRPC command have been removed. This feature is no more available throguh

Reviewer 2: ```suggestion
The `cc.arduino.cli.commands.v1.BoardAttach` gRPC command has been removed. This feature is no longer available through
```

Typo and minor grammar fixes.
